### PR TITLE
{tools}[foss/2023a] pyiron v0.5.1 w/ Python 3.11.3

### PR DESCRIPTION
**Pull request by lcniel**
(created using `eb --new-pr`)

ETA: I have limited time to work on this and did not expect it to be as involved as it turned out to be, it should ideally be split up and merged as separate PR:s. Currently my available time to work on splitting this up is very limited as software building is only one aspect of a position I currently work one day a week at.

### CODE REVIEW

**Comment by lcniel** [comment]
Test report by @lcniel
**FAILED**
Build succeeded for 15 out of 16 (16 easyconfigs in total)
vera-skylake-build - Linux Rocky Linux 8.8, x86_64, Intel Xeon Processor (Skylake, IBRS, no TSX), Python 3.6.8
See https://gist.github.com/lcniel/d8ce2332150c15e3cf0b9c38f12cc0b2 for a full test report.

**Comment by lcniel** [comment]
Test report by @lcniel
**FAILED**
Build succeeded for 37 out of 38 (16 easyconfigs in total)
vera-icelake-build - Linux Rocky Linux 8.8, x86_64, Intel(R) Xeon(R) Silver 4316 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/lcniel/5d2b6731071a2d2cd11dfdd67a4308e7 for a full test report.

**Comment by lcniel** [comment]
Hmm, something broke when doing the from_pr build. Will have to check that later.

**Comment by verdurin** [comment]
@lcniel did you really intend to include all the other PRs in addition to `pyiron`? 
That normally indicates you didn't start from a clean checkout.

**Comment by lcniel** [comment]
> @lcniel did you really intend to include all the other PRs in addition to `pyiron`? That normally indicates you didn't start from a clean checkout.

I did, there's just a ton of dependencies that weren't updated upstream, with a lot of un-updated sub-dependencies... But this is becoming so complex that there's a risk that some of those dependencies will be covered by the time it's finished...

**Comment by casparvl** [comment]
@lcniel you can remove that erronesouly added easyconfig either from command line as described [here](https://docs.easybuild.io/integration-with-github/#github_new_update_pr_delete) or simply delete it in the GitHub web interface.

Also, like @verdurin signaled, it is unusual to have this many easyconfigs in one PR. Generally, I would recommend splitting it up to give the reviewer (like me :)) a bit of an easier job. You can then open a Pyiron PR that depends on other PRs by putting this in the opening post, like was done in e.g. [this PR](https://github.com/easybuilders/easybuild-easyconfigs/pull/19396).

That way, we can process the dependencies in batches of 5 or so at a time. If there is _some_ logical grouping (e.g. a Pyiron dep that itself requires 4 other deps), I'd bunch those together in one PR.

> But this is becoming so complex that there's a risk that some of those dependencies will be covered by the time it's finished... 

It also prevents this to some degree, since even if one of those dependencies has issues, at least the other PRs containing dependencies can already be merged. Finally, it also makes figuring out the CI issues a bit easier, as there are fewer changes per PR that can cause issues :)

Let me know what you prefer. I'm willing to give it a go to review this PR as a single entity (after you've fixed the CI issues) if you feel a bit overwhelmed by having to split it now. But splitting it _may_ lead to a faster merge in the end... :)

**Comment by lcniel** [comment]
> Let me know what you prefer. I'm willing to give it a go to review this PR as a single entity (after you've fixed the CI issues) if you feel a bit overwhelmed by having to split it now. But splitting it may lead to a faster merge in the end... :)

Thanks a lot, then I will try to fix the CI issues. I only work with easybuild-related stuff 20%, so I don't have so much time to spend on this (basically the choice is between trying to fix it and abandoning it). I would never even have entertained taking this PR on if I had anticipated that it would grow like this and require this ridiculous amount of patching, or at least I would have done a much more modest toolchain/version move.

**Comment by lcniel** [comment]
Test report by @lcniel
**FAILED**
Build succeeded for 2 out of 3 (18 easyconfigs in total)
vera-skylake-build - Linux Rocky Linux 8.8, x86_64, Intel Xeon Processor (Skylake, IBRS, no TSX), Python 3.6.8
See https://gist.github.com/lcniel/0a666bf1a7b49ee9907bad88653d9493 for a full test report.

**Comment by casparvl** [comment]
> Thanks a lot, then I will try to fix the CI issues. I only work with easybuild-related stuff 20%, so I don't have so much time to spend on this (basically the choice is between trying to fix it and abandoning it). I would never even have entertained taking this PR on if I had anticipated that it would grow like this and require this ridiculous amount of patching, or at least I would have done a much more modest toolchain/version move.

Well, we definitely appreciate the effort! :) Maybe two tips that might make your life more convenient:

- I see a ton of missing checksum errors in the CI. You could check out the `--inject-checksums` argument the let EasyBuild automatically inject them. I would typically combine it with a `--force-download`, to make sure you have the most up to date checksums

- The CI also detect some style errors. Rather than having to wait for the CI every time to see if you managed to resolve them, you can run the style check locally by running EasyBuild with `--check-style`.

**Comment by casparvl** [comment]
By the way: I'm going on holiday after today. If you manage to get the CI to pass, feel free to ask in the EasyBuild Slack for a review from one of the other maintainers.

**Comment by lcniel** [comment]
Test report by @lcniel
**FAILED**
Build succeeded for 47 out of 48 (19 easyconfigs in total)
vera-skylake-build - Linux Rocky Linux 8.9, x86_64, Intel Xeon Processor (Skylake, IBRS, no TSX), Python 3.6.8
See https://gist.github.com/lcniel/092e5a17fc0bcc3ed1985297c45c6517 for a full test report.

**Comment by lcniel** [comment]
Test report by @lcniel
**SUCCESS**
Build succeeded for 18 out of 18 (18 easyconfigs in total)
vera-skylake-build - Linux Rocky Linux 8.9, x86_64, Intel Xeon Processor (Skylake, IBRS, no TSX), Python 3.6.8
See https://gist.github.com/lcniel/0c1a796b6786faf660afc77ed91add6e for a full test report.

**Comment by lcniel** [comment]
Test report by @lcniel
**SUCCESS**
Build succeeded for 18 out of 18 (18 easyconfigs in total)
vera-skylake-build - Linux Rocky Linux 8.9, x86_64, Intel Xeon Processor (Skylake, IBRS, no TSX), Python 3.6.8
See https://gist.github.com/lcniel/f18c851f50daed815be87e999544839b for a full test report.

**Comment by lcniel** [comment]
Test report by @lcniel
**SUCCESS**
Build succeeded for 17 out of 17 (17 easyconfigs in total)
vera-skylake-build - Linux Rocky Linux 8.9, x86_64, Intel Xeon Processor (Skylake, IBRS, no TSX), Python 3.6.8
See https://gist.github.com/lcniel/510210791ca0c9eb38b687f8469d3343 for a full test report.

File: easybuild/easyconfigs/n/numexpr/numexpr-2.8.7-foss-2023a.eb
Comment (Flamefire) [inline easybuild/easyconfigs/n/numexpr/numexpr-2.8.7-foss-2023a.eb:20]: Why not use `gfbf` as the toolchain to avoid this?

Comment (ocaisa) [reply easybuild/easyconfigs/n/numexpr/numexpr-2.8.7-foss-2023a.eb:20]: I don't think it should even be necessary to include the toolchain as is, EasyBuild should be able to pick up gfbf as a subtoolchain of foss. If you need to explicitly set the toolchain for a dependency it typically means the software is using the wrong toolchain (as it's almost certainly not going to work for a hierarchical module naming scheme).

Having said that, if `numexpr` does not require MPI, then it shouldn't need foss, and should use `gfbf` anyway

Comment (lcniel) [reply easybuild/easyconfigs/n/numexpr/numexpr-2.8.7-foss-2023a.eb:20]: Yeah, I can't remember why exactly I added it to begin with, I think I was probably misdiagnosing an issue somewhere. Right now I don't have time to rebuild and check that it still works, and this one has no special dependencies so it's a good candidate to split off and review separately.

File: easybuild/easyconfigs/p/pyiron/pyiron-0.5.1-foss-2023a.eb
Comment (Flamefire) [inline easybuild/easyconfigs/p/pyiron/pyiron-0.5.1-foss-2023a.eb:48]: This is an issue as it won't work for non-x86 platforms. We have `pydantic` ECs for older toolchains. I updated them to 2023a and will again submit a PR to yours once that is tested.   
Similarly we have ECs for some of the extensions used, such as Greenlet and GitPython. They will be included in the PR I'm preparing. This will avoid rebuilding python packages used by other modules etc.

Comment (lcniel) [reply easybuild/easyconfigs/p/pyiron/pyiron-0.5.1-foss-2023a.eb]: Really nice work, thank you so much. I had PyDantic at least originally as a separate EC, but I couldn't get Rust to build it properly (some Rust version issue AFAIK).

**Review by Flamefire — APPROVED** [review]

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 18 out of 18 (18 easyconfigs in total)
n1505 - Linux RHEL 8.7 (Ootpa), x86_64, Intel(R) Xeon(R) Platinum 8470 (icelake), Python 3.8.13
See https://gist.github.com/Flamefire/5ecaaca10644e55cdce86bd57a053f91 for a full test report.

**Comment by ocaisa** [comment]
@boegelbot please test @ generoso

**Comment by ocaisa** [comment]
@boegelbot please test @ jsc-zen3

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 57 out of 59 (18 easyconfigs in total)
taurusml2 - Linux RHEL 7.6, POWER, 8335-GTX (power9le), 6 x NVIDIA Tesla V100-SXM2-32GB, 440.64.00, Python 2.7.5
See https://gist.github.com/Flamefire/7c48d9d0942470009010d6519aa17410 for a full test report.

**Comment by ocaisa** [comment]
@Flamefire interesting, we are both failing the same test with the same wrong value, but on different archs

**Comment by Flamefire** [comment]
> @Flamefire interesting, we are both failing the same test with the same wrong value, but on different archs

The wrong value is the 2nd one which is slightly different: 0.6608735502232888 & 0.6460767049351539

I opened an issue about this: https://github.com/molmod/tamkin/issues/43

**Comment by Flamefire** [comment]
I split off some sub PRs to make this easier to handle trying to preserve the correct owner:
- https://github.com/easybuilders/easybuild-easyconfigs/pull/19716
- https://github.com/easybuilders/easybuild-easyconfigs/pull/19717
- https://github.com/easybuilders/easybuild-easyconfigs/pull/19718
- https://github.com/easybuilders/easybuild-easyconfigs/pull/19719
- https://github.com/easybuilders/easybuild-easyconfigs/pull/19720

I guess it makes also sense to tackle TAMkin separately given the observed failures.

Once the above PRs are merged merging develop to this PR should make it much smaller already

**Comment by ocaisa** [comment]
@boegelbot please test @ generoso

**Comment by ocaisa** [comment]
@boegelbot please test @ jsc-zen3

File: easybuild/easyconfigs/t/TAMkin/TAMkin-1.2.6-foss-2023a.eb
Comment (Flamefire) [inline easybuild/easyconfigs/t/TAMkin/TAMkin-1.2.6-foss-2023a.eb:19]:
```suggestion
patches = [
    'TAMkin-1.2.6_fix-python38.patch',
    'TAMkin-1.2.6_fix-test_vsa_no_mass.patch',
]
checksums = [
    {'TAMkin-1.2.6.tar.gz': '1bde275a09be91e5241616aaa9fedc60cb359a263f5c5909bb14431c3a4ed5fd'},
    {'TAMkin-1.2.6_fix-python38.patch': '1633d5b24b012f8c4b6731491e4072c819ebbba65574966b7185ecca52eeac9b'},
    {'TAMkin-1.2.6_fix-test_vsa_no_mass.patch': '67d8b8671d7c71123e8ee9a7b9ebea6562e9711edb41555d1926b1b51d549066'},
]
```

**Review by Flamefire — CHANGES_REQUESTED** [review]
- TAMkin needs a patch, see inline change suggestion
- BioPython 1.83 was merged for 2023a: https://github.com/easybuilders/easybuild-easyconfigs/pull/19590
- MDAnalysis 2.7.0 will be available soon: https://github.com/easybuilders/easybuild-easyconfigs/pull/19939

I'd suggest reusing those as dependencies instead of adding another one (although this PR predates the others)

**Comment by ocaisa** [comment]
@boegelbot please test @ jsc-zen3

File: easybuild/easyconfigs/p/pyiron/pyiron-0.5.1-foss-2023a.eb
Comment (lcniel) [reply easybuild/easyconfigs/p/pyiron/pyiron-0.5.1-foss-2023a.eb:32]: Good question. I can't see where the requirement comes from despite searching, and my memory is failing me at this point (the documentation is not helping, either...). Let's just try 3.0.2 (which is the other version I can see on this toolchain).

File: easybuild/easyconfigs/b/Biopython/Biopython-1.81-foss-2023a.eb
Comment (lcniel) [reply easybuild/easyconfigs/b/Biopython/Biopython-1.81-foss-2023a.eb:7]: Done

File: easybuild/easyconfigs/p/PyTables/PyTables-3.8.0-foss-2023a.eb
Comment (Flamefire) [inline easybuild/easyconfigs/p/PyTables/PyTables-3.8.0-foss-2023a.eb:54]: These 2 changes looks random, was this intended?

File: easybuild/easyconfigs/p/PyTables/PyTables-3.8.0-foss-2023a.eb
Comment (ocaisa) [inline easybuild/easyconfigs/p/PyTables/PyTables-3.8.0-foss-2023a.eb:50]: No need for these changes

File: easybuild/easyconfigs/p/pyiron/pyiron-0.5.1-foss-2023a.eb
Comment (Flamefire) [inline easybuild/easyconfigs/p/pyiron/pyiron-0.5.1-foss-2023a.eb:42]: We have an EC for this: `Pint 0.23` -> Move to dependencies

File: easybuild/easyconfigs/p/pyiron/pyiron-0.5.1-foss-2023a.eb
Comment (Flamefire) [inline easybuild/easyconfigs/p/pyiron/pyiron-0.5.1-foss-2023a.eb:55]: -> Dependency? SQLAlchemy-2.0.25-GCCcore-12.3.0.eb

**Review by Flamefire — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/p/PyTables/PyTables-3.8.0-foss-2023a.eb
Comment (lcniel) [reply easybuild/easyconfigs/p/PyTables/PyTables-3.8.0-foss-2023a.eb:54]: Hmm, sure what happened here, maybe there was a merge upstream of a version with only some of these diffs. Don't see any reason not to delete PyTables changes entirely

File: easybuild/easyconfigs/p/pyiron/pyiron-0.5.1-foss-2023a.eb
Comment (lcniel) [reply easybuild/easyconfigs/p/pyiron/pyiron-0.5.1-foss-2023a.eb]: Done

File: easybuild/easyconfigs/p/pyiron/pyiron-0.5.1-foss-2023a.eb
Comment (lcniel) [reply easybuild/easyconfigs/p/pyiron/pyiron-0.5.1-foss-2023a.eb]: Fixed, that seems to also be where dependency on Greenlet comes from, so that should be pulled in automatically.

File: easybuild/easyconfigs/p/PyTables/PyTables-3.8.0-foss-2023a.eb
Comment (lcniel) [reply easybuild/easyconfigs/p/PyTables/PyTables-3.8.0-foss-2023a.eb]: Yeah, no idea what happened there, maybe a merge of a similar version upstream.

File: easybuild/easyconfigs/p/pyiron/pyiron-0.5.1-foss-2023a.eb
Comment (Flamefire) [inline easybuild/easyconfigs/p/pyiron/pyiron-0.5.1-foss-2023a.eb:36]:
```suggestion
    ('SQLAlchemy', '2.0.25'),
```

Keeping the comma at the end allows to add/remove lines without touching other lines (see your previous commit)

Comment (lcniel) [reply easybuild/easyconfigs/p/pyiron/pyiron-0.5.1-foss-2023a.eb:36]: Good catch!

I found another similar omission. Let's see if my test install succeeds this time...

**Comment by lcniel** [comment]
Test report by @lcniel
**SUCCESS**
Build succeeded for 24 out of 24 (3 easyconfigs in total)
alvis10-11 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) Gold 6338 CPU @ 2.00GHz, Python 3.6.8
See https://gist.github.com/lcniel/42f41ccc7d658aaf4ec58e67bdd9318e for a full test report.

**Comment by ocaisa** [comment]
@boegelbot please test @ generoso

**Comment by ocaisa** [comment]
@boegelbot please test @ jsc-zen3

File: easybuild/easyconfigs/p/pyiron/pyiron-0.5.1-foss-2023a.eb
Comment (ocaisa) [inline easybuild/easyconfigs/p/pyiron/pyiron-0.5.1-foss-2023a.eb:99]:
```suggestion
            {%(name)s-%(version)s.tar.gz': 'bd5af29a935dacbee743c3cc0bcc799e24e7c2483801f56cb9092a79f016f2ed'},
```
This is what I get when I download the file...

**Comment by ocaisa** [comment]
@boegelbot please test @ generoso

**Comment by ocaisa** [comment]
@boegelbot please test @ jsc-zen3

**Review by ocaisa — APPROVED** [review]
@lcniel The merging of this PR is the EasyBuild gift to you for finishing your PhD 😛

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
n1455 - Linux RHEL 8.7 (Ootpa), x86_64, Intel(R) Xeon(R) Platinum 8470 (icelake), Python 3.8.13
See https://gist.github.com/Flamefire/60d5b5bcec35a2334eee76d47de9168a for a full test report.

**Pull request merged**
